### PR TITLE
Add additional benchmarks

### PR DESCRIPTION
**Pull request by cleaton**
Benchmarks used to verify https://github.com/rusterlium/rustler/pull/482 . At that time there was no benchmarks setup in rustler yet so I didn't include them in the PR. 
As there is now, this might be of use for others or CI/CD integration.

### CODE REVIEW

**Comment by evnu** [comment]
Thanks! Can you fix clippy? Also, would be nice to see some example results, could you simply post the machine you used and the output of the benchmark here? :) I wonder what a good structure would be find "interesting" benchmarks if the set of benchmarks in the repository increases. Maybe we should have some logical structure w.r.t. what benchmarks we have.

**Comment by cleaton** [comment]
As an addition, it would be nice if rustler also had a standard way for rustler projects to add benchrmarks/tests that would automatically be setup for PGO, following these steps: https://doc.rust-lang.org/rustc/profile-guided-optimization.html#a-complete-cargo-workflow

Something that could be configured in project Native.ex:
```
use Rustler,
    pgo_run: <Project>.Benchmark.run()

**Comment by evnu** [comment]
> What about having a single Benchmark.run() that can read command line arguments and generate the benchee options? That way for local testing developers can run a single specific test without commenting code while at the same time there's a single place for CI/CD job to call to run all benchmarks as part of some regression job 

W.r.t. regression in CI/CD, in my experience, that seldomly works well unless one puts in considerable effort. CI/CD is run on non-dedicated machines, so results can vary wildly. I wouldn't put too much work into that.

Enabling running all benchmarks with a single command sounds good. I guess for that we would need to make the benchmarks coherent, so that the runs "tell a story". Otherwise, someone running the benchmarks still needs to read the benchmark code first to understand what it is all about.

> As an addition, it would be nice if rustler also had a standard way for rustler projects to add benchrmarks/tests that would automatically be setup for PGO,

Interesting suggestion. This goes out-of-scope of this PR, maybe create a separate ticket just for that? I have never played with PGO yet, so I don't know how complicated it is to use.

File: rustler_benchmarks/native/benchmark/src/nif_various.rs [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    List(Vec<String>),
+    Map(HashMap<i64, String>),
+    String(String),
+    Int(i32),
+    TestStruct(TestStruct),
+}
+
+#[rustler::nif]
+pub fn decode_term(_input: Term) -> NifResult<bool> {
+    Ok(true)
```
[the remark this answers is not in the document]
Comment (cleaton) [reply rustler_benchmarks/native/benchmark/src/nif_various.rs]: It's a good point. I've updated to add some logic to prevent compiler from removing code.

**Comment by evnu** [comment]
@cleaton seems I kind make the github actions run again. Can you rebase on master so that the tests run again?

**Pull request merged**